docs/scripts/link-checker.ts

```diff
@@ -0,0 +1,101 @@
+import https from 'https';
+import http from 'http';
+import process from 'process';
+import { JSDOM } from 'jsdom';
+import { VALIDATED_LINKS } from '../cypress/data/validatedLinks';
+import { sitePaths } from '../src/data/sitePaths';
+
+setTimeout(() => {
+  console.log('This will still run.');
+}, 500);
+
+if (![3, 4].includes(process.argv.length)) {
+  console.error('Expected 3 or 4 arguments!');
+  process.exit(1);
+}
+
+const start = process.argv[2];
+const end = process.argv[3];
+const testPaths = end ? sitePaths.slice(+start, +end) : sitePaths.slice(+start); // Divide the sitePaths array to prevent the socket hang up issue.
+
+testPaths.forEach(async (path, idx) => {
+  await checkPage(path, idx);
+});
+
+async function checkPage(pageUrl, pathIdx) {
+  const request = await http
+    .get(pageUrl, (response) => {
+      let data = '';
+
+      console.log(
+        `🧪[TESTING...] page ${pathIdx} ${pageUrl} statusCode: `,
+        response.statusCode
+      );
+      // A chunk of data has been received.
+      response.on('data', (chunk) => {
+        data += chunk;
+      });
+
+      // The whole response has been received. Print out the result.
+      response.on('end', () => {
+        const dom = new JSDOM(data);
+        dom.window.document.querySelectorAll('a').forEach(async (el) => {
+          await checkURL(el.href, el.tagName, el.text, pageUrl);
+        });
+      });
+    })
+    .on('error', (err) => {
+      console.log('Error: ' + err.message);
+    });
+  request.end();
+}
+
+async function checkURL(urlOrPath, tagName, tagText, pageUrl) {
+  const baseURL = 'http://localhost:3000';
+  if (VALIDATED_LINKS.includes(urlOrPath)) {
+    console.log(
+      `⏭[SKIPPING...] ${urlOrPath} from ${tagName} tag "${tagText}" on ${pageUrl}, because it's pre-validated.`
+    );
+  } else if (urlOrPath.includes('https')) {
```

Comment:
> I don't understand this conditional, why are we checking for https? and why are we replacing about:blank below? are we not going to handle http urls if we have them?

Only external links are `https`, the internal links on localhost are `http`. `about:blank` is somehow injected to the link path, which has to be removed

Reply:
Wouldn't it be cleaner to check for the baseurl then?
```
if (urlOrPath.startsWith(baseURL)) {
// internal link
```

Reply:
It actually doesn't start with `baseURL` because it's just a path, e.g. `/connected-components/authenticator/customization#headers--footers` so we have to add `baseURL` in front of it